pkg/monitoring/rules/rules.go

```diff
@@ -1,180 +1,56 @@
-	"k8s.io/apimachinery/pkg/util/intstr"
-	"k8s.io/utils/ptr"
-)
 
-const (
-	severityAlertLabelKey     = "severity"
-	healthImpactAlertLabelKey = "operator_health_impact"
-	partOfAlertLabelKey       = "kubernetes_operator_part_of"
-	partOfAlertLabelValue     = "kubevirt"
-	componentAlertLabelKey    = "kubernetes_operator_component"
-	componentAlertLabelValue  = "ssp-operator"
+	"kubevirt.io/ssp-operator/pkg/monitoring/rules/alerts"
+	"kubevirt.io/ssp-operator/pkg/monitoring/rules/recordingrules"
 )
 
 const (
-	CommonTemplatesRestoredIncreaseQuery   = "sum(increase(kubevirt_ssp_common_templates_restored_total{pod=~'ssp-operator.*'}[1h]))"
-	TemplateValidatorRejectedIncreaseQuery = "sum(increase(kubevirt_ssp_template_validator_rejected_total{pod=~'virt-template-validator.*'}[1h]))"
+	ruleName             = "prometheus-k8s-rules-cnv"
+	PrometheusKey        = "prometheus"
+	PrometheusValue      = "k8s"
+	RoleLabelKey         = "role"
+	RoleLabelValue       = "alert-rules"
+	KubevirtLabelKey     = "kubevirt.io"
+	KubevirtLabelValue   = "prometheus-rules"
+	PrometheusLabelKey   = "prometheus.ssp.kubevirt.io"
+	PrometheusLabelValue = "true"
 )
 
-// RecordRulesDesc represent SSP Operator Prometheus Record Rules
-type RecordRulesDesc struct {
-	Name        string
-	Expr        intstr.IntOrString
-	Description string
-	Type        string
-}
+func SetupRules() error {
+	err := recordingrules.Register()
+	if err != nil {
+		return err
+	}
 
-// recordRulesDescList lists all SSP Operator Prometheus Record Rules
-var recordRulesDescList = []RecordRulesDesc{
-	{
-		Name:        "kubevirt_ssp_operator_up",
-		Expr:        intstr.FromString("sum(up{pod=~'ssp-operator.*'}) OR on() vector(0)"),
-		Description: "The total number of running ssp-operator pods",
-		Type:        "Gauge",
-	},
-	{
-		Name:        "kubevirt_ssp_template_validator_up",
-		Expr:        intstr.FromString("sum(up{pod=~'virt-template-validator.*'}) OR on() vector(0)"),
-		Description: "The total number of running virt-template-validator pods",
-		Type:        "Gauge",
-	},
-	{
-		Name:        "kubevirt_ssp_operator_reconcile_succeeded_aggregated",
-		Expr:        intstr.FromString("sum(kubevirt_ssp_operator_reconcile_succeeded)"),
-		Description: "The total number of ssp-operator pods reconciling with no errors",
-		Type:        "Gauge",
-	},
-	{
-		Name:        "kubevirt_ssp_template_validator_rejected_increase",
-		Expr:        intstr.FromString(TemplateValidatorRejectedIncreaseQuery + " OR on() vector(0)"),
-		Description: "The increase in the number of rejected template validators, over the last hour",
-		Type:        "Gauge",
-	},
-	{
-		Name:        "kubevirt_ssp_common_templates_restored_increase",
-		Expr:        intstr.FromString(CommonTemplatesRestoredIncreaseQuery + " OR on() vector(0)"),
-		Description: "The increase in the number of common templates restored by the operator back to their original state, over the last hour",
-		Type:        "Gauge",
-	},
+	err = alerts.Register()
```

Comment:
Can you return this function directly?

Reply:
I don't think so since it also register recording rules

Reply:
What do you mean? It also register the recording rules

Reply:
I mean you can change block:
```go
err = alerts.Register()
if err != nil {
    return err
}
return nil
```

To:
```go
return alerts.Register()
```

Reply:
Ohhh ok sure sorry

Reply:
@akrejcir I'm sorry I can't do that, since we do both it don't let me return both directly..
and if we do it only for alerts this not consistent. I prefer to leave as is.
I also inline the error like @0xFelix asked.

Reply:
ok, nevermind. It was just a nit.